### PR TITLE
[TF] Align the optimizer section of the configuration file

### PR DESCRIPTION
**Pull request by evgeniya-egupova**

### CODE REVIEW

**Comment by evgeniya-egupova** [comment]
Jenkins please retry a build

**Review by alexsu52 — COMMENTED** [review]
Could you please add a description of what has been done in this PR.

**Comment by evgeniya-egupova** [comment]
- Aligned optimizers naming
- Created two new schedulers: StepLearningRate and MultiStepLearningRate as analogues to MultiStepLR and StepLR in PyTorch
- Renamed `scheduler_params` to `schedule_params` to match `schedule_type`
- Added optimizers with weight decay
- Added support to define scheduler params directly in "optimizer" section of config

**Comment by evgeniya-egupova** [comment]
Schedule usage examples. Note that defining parameters in "schedule_params" is only supported in TF:
```
"optimizer": {
        "type": "Adam",
        "schedule_type": "piecewise_constant",
        "schedule_params": {
            "boundaries": [15, 20, 30],
            "values": [1e-3, 1e-4, 1e-5, 1e-6]
        }
}
```
```
"optimizer": {
        "type": "Adam",
        "schedule_type": "step", (default)
        "schedule_params": {
            "base_lr": 1e-3,
            "step": 5,   (default: 1)
            "gamma": 0.1 (default: 0.1)
        }
}
```
```
"optimizer": {
        "type": "Adam",
        "schedule_type": "multistep",
        "schedule_params": {
            "base_lr": 1e-3,
            "steps": [10, 15, 30],
            "gamma": 0.1 (default: 0.1)
        }
}
```
```
"optimizer": {
        "type": "Adam",
        "schedule_type": "exponential",
        "schedule_params": {
            "base_lr": 1e-3,
            "step": 5,   (default: 1)
            "gamma": 0.1 (default: 0.1)
        }
}
```
It is also possible to define schedule params in the "optimizer" section. PyTorch supports only this type:
```
"optimizer": {
        "type": "Adam",
        "schedule_type": "multistep",
        "base_lr": 1e-3,
        "steps": [10, 15, 30],
        "gamma": 0.1
}
```

Available schedules:
in PT: 'multistep', 'step', 'exponential' (doesn't support `step` parameter), 'poly', 'plateau'
in TF: 'multistep', 'step', 'exponential', 'piecewise_constant'

**Comment by evgeniya-egupova** [comment]
Optimizers usage example:
```
"optimizer": {
        "type": "Adam" (or "AdamW" only in TF),
        "weight_decay": 0.01 (default: for Adam: 0.0, for AdamW: 0.01)
        "optimizer_params":{
            "betas": [0.9, 0.999], (default)
            "eps": 1e-7,    (default: in TF: 1e-7, in PT: 1e-8)
            "amsgrad": false (default)
        }
        "schedule_type": "step",
        "base_lr": 1e-3
}
```
```
"optimizer": {
        "type": "sgd", (or "momentum" only in TF)
        "weight_decay": 0.01, (default: 0.0)
        "optimizer_params":{
            "momentum": 0.9, (default: for "sgd": 0.0, for "momentum": 0.9)
            "nesterov": false (default)
        }
        "schedule_type": "step",
        "base_lr": 1e-3
}
```
```
"optimizer": {
        "type": "rmsprop",
        "optimizer_params":{
            "rho": 0.9, (default)
            "momentum": 0.9, (default)
            "epsilon": 1e-7 (default)
        }
        "schedule_type": "step",
        "base_lr": 1e-3
}
```
Available optimizers:
in PT: Adam, SGD
in TF: Adam, SGD,  AdamW, momentum, rmsprop

**Comment by alexsu52** [comment]
@evgeniya-egupova  Can I use the PyTorch optimizer config in the TF and vice versa for the same optimizer and schedulers? What needs to be changed to make this possible?

**Review by vshampor — APPROVED** [review]

**Comment by vshampor** [comment]
> @evgeniya-egupova Can I use the PyTorch optimizer config in the TF and vice versa for the same optimizer and schedulers? What needs to be changed to make this possible?

Note that this concerns *sample* configuration file formatting, not NNCFConfig formatting. Sample configuration file formatting is not part of NNCF's public API, so no major efforts should be put into sample-specific config portions interchangeability. If it does not take a lot of effort, however, then it could be done, of course.

**Comment by evgeniya-egupova** [comment]
@alexsu52 @vshampor The main thing that needs to be done so that TF configs could be used in PyTorch is to move the schedule parameters to the `schedule_params` section. Now, some schedule parameters are defined in the `schedule_params`, others directly in the `optimizer` section. This change doesn't seems to be very time consuming, so if there are no other objections, I suggest doing it.

**Review by AlexKoff88 — APPROVED** [review]
